onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -25,23 +25,22 @@ OpenVINOExecutionProvider::OpenVINOExecutionProvider(const OpenVINOExecutionProv
   } else {
     openvino_ep::BackendManager::GetGlobalContext().num_of_threads = info.num_of_threads_;
   }
-  if (info.device_id_ != "") {
-    bool device_found = false;
-    auto available_devices = openvino_ep::BackendManager::GetGlobalContext().ie_core.GetAvailableDevices();
-    for (auto device : available_devices) {
-      if (device == info.device_id_) {
-        device_found = true;
-        break;
-      }
+
+bool device_found = false;
+auto available_devices = openvino_ep::BackendManager::GetGlobalContext().ie_core.GetAvailableDevices();
+  for (auto device : available_devices) {
+    if (device == info.device_type_) {
+      device_found = true;
+      break;
     }
```

Comment:
Maybe add some comment here on the new behaviour

Reply:
added comments to branch - https://github.com/intel/onnxruntime/blob/hdgx_onnx_devicecheck/
The existing pull request should handle this comments as well when merged

Reply:
commit details - https://github.com/intel/onnxruntime/pull/125/commits/7439d309c6696cfc04a402ef1e517a038b7ddca5